torchqdynamics/solvers/mesolve.py

```diff
@@ -0,0 +1,36 @@
+from dataclasses import dataclass
+from typing import List
+
+import torch
+
+
+@dataclass
+class MESolverOption:
```

Comment:
I think this should not be a `MESolverOption` but a generic `SolverOption` dataclass. The solver parameters are independent of the fact that `mesolve` or `sesolve` or `smesolve` is called. Then, each `SolverOption` instance should be dispatched to the appropriate function-dependent `QSolver` if we follow the logic of https://github.com/PierreGuilmin/torchqdynamics/pull/12.